packages/synapse-react-client/src/mocks/msw/handlers/chatHandlers.ts

```diff
@@ -47,6 +52,20 @@ export const getChatbotHandlers = (
     mockAgentChatResponse,
     backendOrigin,
   ),
+
+  //trace events
+  rest.post(
+    `${backendOrigin}${AGENT_CHAT_TRACE(':id')}`,
+    async (_req, res, ctx) => {
+      //mock showing progress (increasing number of items)
+      traceCallCount++
+      if (traceCallCount == 1) {
+        return res(ctx.status(201), ctx.json(mockTraceEventsResponse1))
+      } else if (traceCallCount == 2) {
+        return res(ctx.status(201), ctx.json(mockTraceEventsResponse2))
+      } else return res(ctx.status(201), ctx.json(mockTraceEventsResponse3))
+    },
+  ),
```

Comment:
With react-query (and probably also strictmode), we can't guarantee only one request is made before we should expect to see more data. However, I'm not sure if there's an easy way to change this, so I think this is OK for tests